File: 3st/Hyeonseo/programmer/기사단원의 무기.py
import math

def solution(number, limit, power):
    answer = 0
    
    for num in range(1, number + 1):
        div = 0
        for i in range(1, int(math.isqrt(num)) + 1):
            if num % i == 0:
                if i == num // i: 
                    div += 1
                else:
                    div += 2      
                    
        if div > limit:
            answer += power
        else:
            answer += div
            
    return answer

File: 4st/Eunseong/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        int numbers[] = new int[number + 1];
        for (int i = 1; i <= number; i++) {
            for (int j = i; j <= number; j += i) {
                numbers[j]++;
            }
        }
        for (int i : numbers) {
            answer += (i > limit) ? power : i;
        }
        return answer;
    }
}